boot/zephyr/Kconfig

```diff
@@ -653,12 +658,66 @@ config BOOT_ENCRYPTION_SUPPORT
 	help
 	  Hidden option used to check if image encryption is supported.
 
-config BOOT_ENCRYPT_IMAGE
-	bool "Support for encrypted image updates"
-	depends on BOOT_ENCRYPTION_SUPPORT
+config BOOT_ENCRYPT_IMAGE_WITH_EMBEDDED_KEY
+	bool "Use key that is already on board with MCUboot"
+	depends on BOOT_ENCRYPT_IMAGE
+	help
+	  The key is supposed to be either compiled in or on board.
+	  User is responsible for providing boot_enc_take_key
+	  function that will be able to retrieve the key.
+
+if BOOT_ENCRYPT_IMAGE_WITH_EMBEDDED_KEY
+
+choice  BOOT_ENCRYPT_IMAGE_EMBEDDED_KEY_PROVIDER
+	prompt "Embedded AES key provider"
+	default BOOT_ENCRYPT_IMAGE_GENERATE_BASIC_KEY_PROVIDER
+
+config BOOT_ENCRYPT_IMAGE_GENERATE_BASIC_KEY_PROVIDER
+	bool "Generate basic boot_enc_take_key"
+	depends on BOOT_ENCRYPT_IMAGE_WITH_EMBEDDED_KEY
+	help
+	  Basic implementation of boot_enc_take_key will be implemented,
+	  that will have single key built in, used for all images and
+	  slots.
+
+config BOOT_ENCRYPT_IMAGE_USE_CUSTOM_KEY_PROVIDER
+	bool "User provides source code for key provider"
+
+endchoice # BOOT_ENCRYPT_IMAGE_EMBEDDED_KEY_PROVIDER
+
+config BOOT_ENCRYPT_IMAGE_EMBEDDED_RAW_KEY
+	string "Hexadecimal string representing AES key"
+	depends on BOOT_ENCRYPT_IMAGE_GENERATE_BASIC_KEY_PROVIDER
+	help
+	  AES key in form of hexadecimal string that will be used to
+          generate boot_enc_take_key function, returning the key for
+          decryption and encryption of image.
+	  The key character length should be the double of expected
+	  AES key length in bytes.
+
+config BOOT_ENCRYPT_IMAGE_EMBEDDED_CUSTOM_KEY_PROVIDER_DIR
```

Comment:
User need to modify project on there own anywaey, In a fork or maybe in the feature in this CMakeList.stxt. No better to instruct user to include relevant sources on their own?

Reply:
Removed